gcb/release/cloudbuild.yaml

```diff
@@ -21,7 +21,7 @@ steps:
   dir: "go/src/k8s.io"
   args:
   - "clone"
-  - "--branch=${_TOOL_BRANCH}"
+  - "--branch=${_TOOL_REF}"
   - "https://github.com/${_TOOL_ORG}/${_TOOL_REPO}"
 
```

Comment:
One possibility to checkout a ref would be to insert a new GCB stage here that simply does `git checkout SHA` when TOOL_REF is a string that looks like a sha or sha+branch slug. It should be fairly quick as the git container is already cached and is one of the official GCB builders anyway.

The other option would be to change the clone stage, modify the entry point to sh/bash, and call git twice from the shell (clone and then checkout).

Reply:
thanks for the feedback, I was unsure about that, I will ping you in slack to discuss if you can offcourse